### PR TITLE
LPS-89103 data record collection permissions

### PR DESCRIPTION
**Pull request by rbohl**
Ready for review: https://issues.liferay.com/browse/LPS-89103, cc @marcelabc

### CODE REVIEW

File: developer/frameworks/articles/data-engine/02-data-collection-api/00-intro.markdown [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Data Collections API
+<!--Should maybe describe the details of what each operation is for in this
+article instead of in the next one on building each request, which is getting
+long-->
+
+Data Record Collections (`DEDataRecordCollection`s) are used to collect
+`DEDataRecord`s. A Data Definition defines the fields present in the Data
+Collection, while the Data Record Collection forms a boundary, collecting only
```
Comment (marcelabc) [inline developer/frameworks/articles/data-engine/02-data-collection-api/00-intro.markdown]: A Data Definition defines the fields present in the Data Record, ...

File: developer/frameworks/articles/data-engine/02-data-collection-api/01-data-record-collection-operations.markdown [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+- `String[] roleNames`
+
+Optional parameters: none
+
+Optional builder methods:
+
+- `allowDelete()` to grant deletion permission
+- `allowUpdate()` to grant update permission
+- `allowView()` to grant view permission
+- `allowAddDataRecord()` to grant add permissions, to add `DEDataRecord`
```
Comment (marcelabc) [inline developer/frameworks/articles/data-engine/02-data-collection-api/01-data-record-collection-operations.markdown]: remove the comma, please :)

**Comment by marcelabc** [comment]
I just reviewed and it is fine for me 👍

**Comment by rbohl** [comment]
Thanks! I'm about to start writing up the other data record collection operations. Is it okay to just push the commits into my branch and update this PR?

@marcelabc

**Comment by rbohl** [comment]
@marcelabc This is ready for review. I'll put the tickets pertaining to the new information in review now.

File: developer/frameworks/articles/data-engine/02-data-collection-api/01-data-record-collection-operations.markdown [hunk 3]
```diff
@@ -152,6 +152,14 @@ To delete a `DEDataRecordCollection`, pass a `DEDataRecordCollectionDeleteReques
 Required parameter: `long deDataRecordCollectionId`
 Optional parameters: none
 
+## Deleting a Data Record
+
+To delete a `DEDataRecordCollection`, pass a `DEDataRecordCollectionDeleteRequest` to
```
Comment (marcelabc) [inline developer/frameworks/articles/data-engine/02-data-collection-api/01-data-record-collection-operations.markdown]: "To delete a **DEDataRecord**, pass a **DEDataRecordCollectionDeleteRecordRequest** to ..

File: developer/frameworks/articles/data-engine/02-data-collection-api/01-data-record-collection-operations.markdown [hunk 4]
```diff
@@ -152,6 +152,14 @@ To delete a `DEDataRecordCollection`, pass a `DEDataRecordCollectionDeleteReques
...
+`DEDataRecordCollectionService.execute()`.
+
+Required parameter: `long deDataRecordCollectionId`
```
Comment (marcelabc) [inline developer/frameworks/articles/data-engine/02-data-collection-api/01-data-record-collection-operations.markdown:163]: Required parameter: long **deDataRecordId**

**Comment by marcelabc** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by rbohl** [comment]
@gabrielwas @marcelabc I pushed up sections on building requests for:

listing data records -- https://github.com/gabrielwas/liferay-docs/pull/4/commits/2c1b6ac0a6eb49129aaf541f2c2b0b3752b94bc3

searching for data collections -- https://github.com/gabrielwas/liferay-docs/pull/4/commits/155367757ce5a933d5495b11214a1d1ceaf9ea16

**Pull request closed without merging**